src/ccache.cpp

```diff
@@ -2690,7 +2769,7 @@ cc_process_args(struct args* args,
 
     // Debugging is handled specially, so that we know if we can strip line
     // number info.
-    if (str_startswith(argv[i], "-g")) {
+    if (str_startswith(argv[i], "-g") && guessed_compiler != GUESSED_MSVC) {
```

Comment:
MSVC has a `-/GZ` parameter, which would be taken into consideration here. But since the comparison is case sensitive I think it should be fine without it.

I'll do a test and update the PR.